lib/request_cache/middleware.ex

```diff
@@ -33,6 +44,14 @@ if RequestCache.Application.dependency_found?(:absinthe) and RequestCache.Applic
         resolution
       end
     end
+
+    defp ensure_valid_ttl(opts) do
+      case Keyword.get(opts, :ttl) do
+        ttl when is_integer(ttl) ->
+          opts
+        _ ->
+          Keyword.merge(opts, [ttl: RequestCache.Config.default_ttl()])
+      end
```

Comment:
This would probably be more concise as:
```elixir
ttl = opts[:ttl] || RequestCache.Config.default_ttl()
Keyword.merge(opts, [ttl: ttl])
```

Reply:
Keyword.put_new

Reply:
put_new doesn't work, the key is there already...